modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -118,8 +127,32 @@ module ChapelIteratorSupport {
       return _getIteratorZipInternal(x, 1);
   }
 
+  inline proc _getIteratorZip(type t: _tuple) {
+    if (!(isTupleType(t))) then
+      compilerError("cannot zip over a type that is not a homogeneous tuple");
+    else if (!(isEnumType(t(1)))) then
+      compilerError("cannot zip over a tuple of types that is not of enums");
```

Comment:
I am simplifying the error cases, since after taking a second look some of them don't do what I wanted them to do. I'll add tests for the errors I keep.